### PR TITLE
feat: analytics site integration API spec

### PR DESCRIPTION
**Pull request by iuliag**
After the setup for an analytics solution is completed for the site represented by the base URL, the new hook will be called with the different artifacts details. SpaceCat will use these in its automations such as:
* notifying users about the new reporting available
* opening a PR to embed the data collection scripts into the site
* importing RUM data into the analytics solution (Adobe Analytics or Customer Journey Analytics)

The SpaceCat site id will be used as a shared identifier between SpaceCat and Omnivore, as it's more stable than the baseURL.
SpaceCat will initially start using `siteId`, `dataCollection` and `reporting`, as required by the Slack-based flow.
Implementation will be done via https://github.com/adobe/spacecat-api-service/issues/237

### CODE REVIEW

File: docs/openapi/schemas.yaml [hunk 1]
```diff
@@ -634,3 +634,135 @@ RUMDomainDiscoveryAlert:
...
+      description: |
+        The configuration for the data collection to embed on the site.
+        The keys represent the environments (development, staging, production).
+        The values represent the URL or ID of the data collection script.
+        For Adobe Analytics and Customer Journey Analytics, the values are either
+        URL for the Adobe Launch script or the datastream ID.
+      type: object
+      additionalProperties:
+        type: string
+      require:
```
Comment (solaris007) [inline docs/openapi/schemas.yaml:660]: should be `required`

File: docs/openapi/schemas.yaml [hunk 2]
```diff
@@ -634,3 +634,135 @@ RUMDomainDiscoveryAlert:
...
+    - baseURL
+    - reporting
+    - dataCollection
+    - dataStore
+    - dataMapping
+  properties:
+    baseURL:
+      description: The base URL of the site
+      $ref: '#/URL'
+    dataCollection:
```
Comment (solaris007) [inline docs/openapi/schemas.yaml:650]: in dataCollection, environment keys (development, staging, production) are mentioned, but there's no enforcement mechanism (like enum) to restrict values to these keys only. While additionalProperties allows flexibility, using enum for known keys can help validate configurations against expected environments, unless it is intended to allow arbitrary properties, which then should be documented.

**Review by solaris007 — APPROVED** [review]

File: docs/openapi/schemas.yaml (hunk 2 above)
Comment (iuliag) [reply docs/openapi/schemas.yaml]: It's intended to have arbitrary naming for the environments, except for production, hence I added that one as required.

**Comment by iuliag** [comment]
@solaris007 , @alinarublea and @jbranson000 could you please re-review?
I want to conclude the API spec ticket, implementation will follow when prioritized.

**Pull request merged**